src/AppInstallerRepositoryCore/Microsoft/Schema/1_2/Interface_1_2.cpp

```diff
@@ -13,27 +13,37 @@ namespace AppInstaller::Repository::Microsoft::Schema::V1_2
 {
     namespace
     {
+        void AddNormalizedName(const Utility::NameNormalizer& normalizer, const Manifest::string_t& name, std::vector<Utility::NormalizedString>& out)
+        {
+            Utility::NormalizedString value = normalizer.NormalizeName(Utility::FoldCase(name)).Name();
+            if (std::find(out.begin(), out.end(), value) == out.end())
+            {
+                out.emplace_back(std::move(value));
+            }
+        }
+
         void AddLocalizationNormalizedName(const Utility::NameNormalizer& normalizer, const Manifest::ManifestLocalization& localization, std::vector<Utility::NormalizedString>& out)
         {
             if (localization.Contains(Manifest::Localization::PackageName))
             {
-                Utility::NormalizedString value = normalizer.NormalizeName(Utility::FoldCase(localization.Get<Manifest::Localization::PackageName>())).Name();
-                if (std::find(out.begin(), out.end(), value) == out.end())
-                {
-                    out.emplace_back(std::move(value));
-                }
+                AddNormalizedName(normalizer, localization.Get<Manifest::Localization::PackageName>(), out);
+            }
+        }
+
+        void AddNormalizedPublisher(const Utility::NameNormalizer& normalizer, const Manifest::string_t& publisher, std::vector<Utility::NormalizedString>& out)

```

Comment:
>AddNormalizedPublisher [](http://example.com/codeflow?start=13&length=22)

nit: looks like this implementation is same as the AddNormalizedName above, shall we combine them like AddNormalizedValue?